RealmSwift/Projection.swift

```diff
@@ -0,0 +1,1342 @@
+                case .update(let collection,
+                             deletions: let deletions,
+                             insertions: let insertions,
+                             modifications: let modifications):
+                    block(.update(anyCtor(collection),
+                                  deletions: deletions,
+                                  insertions: insertions,
+                                  modifications: modifications))
+                case .error(let error):
+                    block(.error(error))
+                }
+            }
+        }
+
+    /**
+     Returns the object at the given index (get), or replaces the object at the given index (set).
+
+     - warning: You can only set an object during a write transaction.
+
+     - parameter index: The index of the object to retrieve or replace.
+     */
+    public subscript(position: Int) -> NewElement {
+        get {
+            backingList[position][keyPath: keyPath] as! NewElement
+        }
+        set {
+            backingList[position].setValue(newValue, forKeyPath: propertyName)
+        }
+    }
+
+    /// The position of the first element in a non-empty collection.
+    /// Identical to endIndex in an empty collection.
+    public var startIndex: Int {
+        backingList.startIndex
+    }
+    /// The collection's "past the end" position.
+    /// endIndex is not a valid argument to subscript, and is always reachable from startIndex by
+    /// zero or more applications of successor().
+    public var endIndex: Int {
+        backingList.endIndex
+    }
+    /// The Realm which manages the object.
+    public var realm: Realm? {
+        backingList.realm
+    }
+    /// Indicates if the collection can no longer be accessed.
+    public var isInvalidated: Bool {
+        backingList.isInvalidated
+    }
+    /// A human-readable description of the object.
+    public var description: String {
+        var elements = ""
+        for (i, o) in self.enumerated() {
+            elements += "\t[\(i)] \(o)\n"
+        }
+        return "\(type(of: self))<\(Element.self)> <\(self)> {\n" +
+        elements +
+        "}"
+    }
+    /**
+     Returns the index of an object in the linking objects, or `nil` if the object is not present.
+
+     - parameter object: The object whose index is being queried.
+     */
+    public func index(of object: Element) -> Int? {
+        backingList.map({$0[keyPath: self.keyPath] as! Element}).firstIndex(of: object)
+    }
+    public var isFrozen: Bool {
+        backingList.isFrozen
+    }
+    public func freeze() -> Self {
+        anyCtor(backingList.freeze())
+    }
+    public func thaw() -> Self? {
+        guard let backingList = backingList.thaw() else {
+            return nil
+        }
+        return anyCtor(backingList)
+    }
+
+    private var backingList: List<Object>
+    private let keyPath: AnyKeyPath
+    private let propertyName: String
+    private let anyCtor: (List<Object>) -> Self
```

Comment:
There doesn't appear to be any reason for this to be a closure.

Reply:
it's a proper init now